src/basic/fileio.c

```diff
@@ -186,21 +186,7 @@ int read_one_line_file(const char *fn, char **line) {
         if (!f)
                 return -errno;
 
-        if (!fgets(t, sizeof(t), f)) {
-
-                if (ferror(f))
-                        return errno > 0 ? -errno : -EIO;
-
-                t[0] = 0;
-        }
-
-        c = strdup(t);
-        if (!c)
-                return -ENOMEM;
-        truncate_nl(c);
-
-        *line = c;
-        return 0;
+        return read_line(f, LONG_LINE_MAX, line);
```

Comment:
It seems that some callers don't expect that `read_one_line_file` might return something different from `0` and negative values. For example, `bump_rlimit_nofile` should be fixed to prevent the crash mentioned in the comments:
```diff
diff --git a/src/core/main.c b/src/core/main.c
index fbf8876..8a7bf58 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -1187,7 +1187,7 @@ static int bump_rlimit_nofile(struct rlimit *saved_rlimit) {

         /* Get current RLIMIT_NOFILE maximum compiled into the kernel. */
         r = read_one_line_file("/proc/sys/fs/nr_open", &nr_open);
-        if (r == 0)
+        if (r >= 0)
                 r = safe_atoi(nr_open, &min_max);
         /* If we fail, fallback to the hard-coded kernel limit of 1024 * 1024. */
         if (r < 0)
```